core/src/main/java/ai/timefold/solver/core/impl/exhaustivesearch/decider/ExhaustiveSearchDecider.java

```diff
@@ -131,29 +126,32 @@ public void expandNode(ExhaustiveSearchStepScope<Solution_> stepScope) {
     @SuppressWarnings("unchecked")
     private <Score_ extends Score<Score_>> void doMove(ExhaustiveSearchStepScope<Solution_> stepScope,
             ExhaustiveSearchNode moveNode) {
-        InnerScoreDirector<Solution_, Score_> scoreDirector = stepScope.getScoreDirector();
-        Move<Solution_> move = moveNode.getMove();
-        try (var ephemeralMoveDirector = scoreDirector.getMoveDirector().ephemeral()) {
-            move.execute(ephemeralMoveDirector);
-            processMove(stepScope, moveNode);
-            moveNode.setUndoMove(ephemeralMoveDirector.createUndoMove());
-        }
-        // TODO reuse scoreDirector.doAndProcessMove() unless it's an expandableNode
+        var scoreDirector = stepScope.<Score_> getScoreDirector();
+        var move = moveNode.getMove();
+        var undoMove = scoreDirector.getMoveDirector().executeTemporary(move,
```

Comment:
Let's make the name consistent. We have used `executeTemporaryMove` in the CH decider.

Reply:
That was calling a different method on the score director.
A method on "move director" which also has "move" in its name just seems unnecessarily verbose.